components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/RestApiAdminUtils.java

```diff
@@ -55,4 +73,89 @@ public static boolean isBlockConditionAccessibleToUser(String user, BlockConditi
         return !StringUtils.isBlank(blockCondition.getTenantDomain()) && blockCondition.getTenantDomain()
                 .equals(userTenantDomain);
     }
+
+    /**
+     * Extract the content of the provided tenant theme archive
+     *
+     * @param themeFile    content relevant to the tenant theme
+     * @param tenantDomain tenant to which the theme is imported
+     * @throws APIManagementException if an error occurs while importing tenant theme
+     */
+    public static void deployTenantTheme(InputStream themeFile, String tenantDomain) throws APIManagementException {
+
+        ZipInputStream zipInputStream = null;
+        byte[] buffer = new byte[1024];
+
+        String outputFolder = "repository" + File.separator + "deployment" + File.separator + "server"
+                + File.separator + "jaggeryapps" + File.separator + "devportal" + File.separator + "site"
+                + File.separator + "public" + File.separator + "tenant_themes" + File.separator + tenantDomain;
+
+        try {
+            //create output directory if it does not exist
+            File folder = new File(outputFolder);
+            if (!folder.exists()) {
+                if (!folder.mkdirs()) {
+                    APIUtil.handleException("Unable to create tenant theme directory at " + outputFolder);
+                }
+            } else {
+                //remove existing files inside the directory
+                FileUtils.cleanDirectory(folder);
+            }
+
+            //get the zip file content
+            zipInputStream = new ZipInputStream(themeFile);
+            //get the zipped file list entry
+            ZipEntry zipEntry = zipInputStream.getNextEntry();
+
+            while (zipEntry != null) {
+
```

Comment:
Shall we remove the empty lines. Please check other places as well.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9702/commits/e22c474785b0d7f5a6c4c0ded773ac0ad2d03fc5